### DISCUSSION TITLE
Artillery API test for 1VU showing incorrect value for 'median' under 'http.response_time'

### DISCUSSION THREAD

**Discussion by akshatrastogi01** [artilleryio/artillery#3222] [Artillery]
Artillery API test for 1VU showing incorrect value for 'median' under 'http.response_time'.
Detailed report is attached.
The test was done for 1VU on Fargate.

Attributes for 'http.response_time' (ms) are:
1. min - 122
2. max - 122
3. median - 122.7
4. p95 - 122.7
5. p99 - 122.7

In noway the median should come out to be - 122.7

[report.html.zip](https://github.com/user-attachments/files/15936977/report.html.zip)

**Comment by hassy** [comment]
Your test is making a single request, i.e. the data set only has one sample. Percentile & median calculations don't make much intuitive sense in those cases. Artillery uses [DDSketch](https://www.vldb.org/pvldb/vol12/p2195-masson.pdf) under the hood for calculating summary statistics. It's probabilistic data structure and the value you're seeing is the estimate based on a single number.

https://stats.stackexchange.com/a/193813

**Reply by akshatrastogi01** [reply]
Thanks @hassy - so I will take it as approximation deviation, that is also acceptable.  Though going by maths it was not adding up. Appreciate your responding! Cheers! :)
